dpnp/linalg/dpnp_utils_linalg.py

```diff
@@ -999,6 +1000,37 @@ def dpnp_inv(a):
     return b_f
 
 
+def dpnp_matrix_rank(A, tol=None, hermitian=False):
+    """
+    dpnp_matrix_rank(A, tol=None, hermitian=False)
+
+    Return matrix rank of array using SVD method.
+
+    """
+
+    if A.ndim < 2:
+        return (A != 0).any().astype(int)
+
+    S = dpnp_svd(A, compute_uv=False, hermitian=hermitian)
+
+    if tol is None:
+        tol = (
+            S.max(axis=-1, keepdims=True)
+            * max(A.shape[-2:])
+            * dpnp.finfo(S.dtype).eps
+        )
```

Comment:
I guess it would be much more efficient to call scalar multiplication first rather than calling to multiply kernels in a row:
```suggestion
        rtol = max(A.shape[-2:]) * dpnp.finfo(S.dtype).eps
        tol = S.max(axis=-1, keepdims=True) * rtol
```

Reply:
Make sense!
Thanks